samples/cpp/benchmark_app/property_map.hpp

```diff
@@ -0,0 +1,172 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <openvino/runtime/intel_gna/properties.hpp>
+#include <openvino/runtime/intel_gpu/properties.hpp>
+#include <openvino/runtime/intel_gpu/remote_properties.hpp>
+#include <openvino/runtime/intel_myriad/hddl_properties.hpp>
+#include <openvino/runtime/intel_myriad/myriad_properties.hpp>
+
+#define VALUE_2_STRING(name) #name
+#define INSERT_NAME_VALUE_MAP(out_map, value)        \
+    {                                                \
+        std::string str_key = VALUE_2_STRING(value); \
+        std::stringstream str_value;                 \
+        str_value << value;                          \
+        out_map[str_key] = str_value.str();          \
+    }
+
+enum PropertyType {
+    PROPERTY_TYPE_STRING = 0,
+    PROPERTY_TYPE_INT,
+    PROPERTY_TYPE_UNSIGNED_INT,
+    PROPERTY_TYPE_INT64,
+    PROPERTY_TYPE_ENUM,
+    PROPERTY_TYPE_BOOL,
+    PROPERTY_TYPE_FLOAT,
+    PROPERTY_TYPE_STRING_FLOAT_MAP,
+};
+
+struct PropertyInfo {
+    std::string property_name;
+    PropertyType property_type;
+};
+
+#define INSERT_PROPERTY_MAP(out_map, value, type, reversed) \
+    {                                                       \
+        std::string str_key = VALUE_2_STRING(value);        \
+        std::string str_name = value.name();                \
+        PropertyInfo property_info;                         \
+        property_info.property_type = type;                 \
+        if (reversed) {                                     \
+            property_info.property_name = str_key;          \
+            out_map[str_name] = property_info;              \
+        } else {                                            \
+            property_info.property_name = str_name;         \
+            out_map[str_key] = property_info;               \
+        }                                                   \
+    }
+
+void generate_property_map(std::map<std::string, PropertyInfo>& property_map, bool reversed) {
+    INSERT_PROPERTY_MAP(property_map, ov::enable_profiling, PROPERTY_TYPE_BOOL, reversed);
```

Comment:
Do you can image how to support it in the future? Each time when any property is added into the any plugin, it should be updated here.

Reply:
I agree with you, it is a question.
Any good idea on how to resolve it?

Reply:
> I agree with you, it is a question. Any good idea on how to resolve it?

Why you can't use string config file with new properties? For example
```
{
    "GPU": { "INFERENCE_PRECISION_HINT" : "f16" }
}
```

Reply:
totally agree with @nkogteva
In API 2.0 we have methods which accept `AnyMap`:
```c++
CompiledModel compile_model(const std::shared_ptr<const ov::Model>& model, const AnyMap& properties = {});
```
So, they contain string => ov::Any (from string) and plugin can accept this. String internally will be converted to actual type requested by plugin. E.g.
```c++
ov::Any a = "1";
int i = a.as<int>();
```
And it will work.

Reply:
@xufang-lisa let's follow this solution.